lib/sycamore/sycamore/transforms/extract_document_structure.py

```diff
@@ -16,6 +18,50 @@ def extract(document: Document) -> Union[Document, HierarchicalDocument]:
         pass
 
 
+class StructureByImages(DocumentStructure):
+    @staticmethod
+    def extract(doc: Document) -> HierarchicalDocument:
+        doc = HierarchicalDocument(doc.data)
+        doc.data["relationships"] = doc.get("relationships", {})
+        doc.data["label"] = doc.get("label", "DOCUMENT")
+
+        images = [child for child in doc.children if child.type == "Image"]
+        doc.children = []
```

Comment:
wait what if there's other elements?

Reply:
This is only for extracting images, which was datarobots specific use case. They wanted to extract each image as a DAG, and ignore text for now. There is probably a better abstraction that makes this more flexible but I'm not sure how it would look. I tried a different abstraction using functional programming but I was running into serialization issues left and right because of pydantic so I gave up on doing that.